### PR TITLE
Add tests and improve functionality of the Popolo JSON importer

### PR DESCRIPTION
**Pull request by mhl**
These changes are motivated so that we can use the Popolo JSON importer in write-it, but all generally improve the utility and robustness of the importer. This includes:

* Better test coverage
* Test coverage being reported to coveralls.io: https://coveralls.io/github/mysociety/django-popolo
* Use the observer pattern to allow callbacks for when objects are create or updated
* New options for behaviour if Popolo JSON data is too long for the django-popolo model fields

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -40,5 +40,6 @@ def read_file(filename):
     tests_require=['fake-factory'],
     install_requires=[
         "django-model-utils",
+        "six",
```
Comment (dracos) [inline setup.py]: Django has a copy at django.utils.six if you wanted to save a new requires.

File: popolo/importers/popit.py [hunk 2]
```diff
@@ -92,6 +101,29 @@ def get_popolo_model_class(self, model_name):
...
+                max_length = None
+            else:
+                raise
+        if max_length is None:
+            # Then do no more checking, just set the value:
+            setattr(django_object, field_name, value)
+        else:
+            if self.truncate == 'exception':
+                # Just set the value, if it's too long an exception will be raised:
+                setattr(django_object, field_name, value)
```
Comment (dracos) [inline popolo/importers/popit.py]: setattr() can be after this if/else, it's called in both cases.

**Comment by dracos** [comment]
Looks good :)

File: setup.py (hunk 1 above)
Comment (mhl) [inline setup.py]: Good point, I've done that.

File: popolo/importers/popit.py (hunk 2 above)
Comment (mhl) [inline popolo/importers/popit.py]: Right, yes - trying it that way, I think it is clearer with a single `setattr` and a couple of comments.

**Comment by mhl** [comment]
Thanks, @dracos - I've added a few more commits since then, but nothing very significant, I think:

* I made a fixup to remove the check for whether a field has max_length (which was broken anyway), since max_length is defined on `Field` to be `None` unless a subclass changes that.
* Slightly improve the test coverage
* Renaming the importer module and class so they don't sound PopIt-specific

So I'll merge that and release it as version 0.0.6 if that's OK.

**Comment by dracos** [comment]
🎃

**Comment by mhl** [comment]
@dracos I've merged this, but it turns out that I'm not a maintainer of the package on PyPi so I can't release the new version. Could you either do that, or add `mhl20` as a maintainer?

**Pull request merged**